crates/bevy_tasks/src/task_pool.rs

```diff
@@ -79,6 +79,14 @@ impl TaskPoolBuilder {
         self
     }
 
+    /// Sets a callback that is invoked once for every created thread as it starts.
+    ///
+    /// This function is similar to `on_thread_spawn`, but take `Option<Arc<...>>` directly.
+    pub fn on_thread_spawn_raw(mut self, f: Option<Arc<dyn Fn() + Send + Sync + 'static>>) -> Self {
```

Comment:
This is only called once per thread. The overhead versus user facing API complexity tradeoff is not worth it IMO.